src/main/java/com/boomi/flow/services/boomi/mdh/common/Entities.java

```diff
@@ -1,30 +1,75 @@
 package com.boomi.flow.services.boomi.mdh.common;
 
+import com.boomi.flow.services.boomi.mdh.quarantine.QuarantineEntry;
+import com.boomi.flow.services.boomi.mdh.quarantine.QuarantineEntryConstants;
 import com.manywho.sdk.api.run.elements.type.MObject;
 import com.manywho.sdk.api.run.elements.type.Property;
 
 import java.util.List;
 import java.util.Map;
+import java.util.UUID;
 import java.util.stream.Collectors;
 
 public class Entities {
-    public static MObject createEntityMObject(String id, String type, Map<String, Map<String, Object>> entity) {
+
+    public static MObject createGoldenRecordMObject(String universeId, String id, Map<String,  Map<String, Object>> entity) {
         if (entity == null || entity.isEmpty()) {
             return null;
         }
 
         var entry = entity.entrySet().iterator().next();
+        List<Property> properties = createPropertiesModel(id, entry.getValue());
+
+        return new MObject(universeId + " golden-record", id, properties);
+    }
+
+    public static MObject createQuarantineMObject(String universeId, QuarantineEntry entry, Map<String,  Map<String, Object>> entity) {
+        if (entity == null || entity.isEmpty()) {
+            return null;
+        }
+
+        var entry1 = entity.entrySet().iterator().next();
```

Comment:
There should be only one entity for the quarantine entry. I just use a Map because I don't know the name of the entity.
I am removing the last parameter anyways as it can be obtained from entry.
````
   <QuarantineEntry createdDate="2014-07-12T21:45:54Z" sourceId="salesforce" sourceEntityId="2" transactionId="01234567-89ab-cdef-0123-456789abcdef">
      <cause>ENRICH_ERROR</cause>
      <reason>At data quality step 'boomi-internal-sitest-priv_dev:::AddressVerification': [304] Address Not Found.</reason>
      <entity>
         <contact>
            <id>2</id>
            <name>jack</name>
            <address_1>not home</address_1>
            <city>berwyn</city>
            <state>PA</state>
            <zip>19312</zip>
         </contact>
      </entity>
   </QuarantineEntry>
````